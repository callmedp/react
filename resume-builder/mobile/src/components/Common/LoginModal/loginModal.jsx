import React from 'react';
import Modal from 'react-modal';
import './loginModal.scss'


if (typeof document !== 'undefined') {
    Modal.setAppElement(document.getElementById('react-app'));
}


export default class LoginModal extends React.Component {

    constructor(props) {
        super(props);
        this.staticUrl = (window && window.config && window.config.staticUrl) || '/media/static/';
        this.closeModal = this.closeModal.bind(this);
        this.handleEmailInput = this.handleEmailInput.bind(this);
        this.handlePasswordInput = this.handlePasswordInput.bind(this);
        this.handleLogin = this.handleLogin.bind(this);
        this.state = {
            email: "",
            password: "",
            error: false,
            errorMessage: ""
        }
    }

    handleEmailInput(event) {
        this.setState({
            "email": event.target.value
        })
    }

    handlePasswordInput(event) {
        this.setState({
            "password": event.target.value
        })
    }

    async handleLogin(event) {
        event.preventDefault();
        const { history } = this.props, templateId = localStorage.getItem('selected_template') || ''
        this.setState({
            "error": false,
            "errorMessage": ''
        })
        if (!this.state.email || !this.state.password) {
            let errorMessage ="";
            if(! this.state.email && !this.state.password){
                errorMessage = "Email and Password is required."
            }
            else if (!this.state.email){
                errorMessage = "Email is required."
            }
            else {
                errorMessage = "Password is required"
            }
            this.setState({
                'error': true,
                "errorMessage": errorMessage
            })
            return;
        }
        try {
            const result = await this.props.loginCandidate({ email: this.state.email, password: this.state.password }, history, true)
            if (templateId) {
                localStorage.setItem('selected_template', templateId);
            }
            this.closeModal();
            this.props.handleLoginSuccess()
        }
        catch (e) {
            this.setState({
                error: true,
                errorMessage: e.message
            })

        }
    }
    closeModal() {
        this.props.hideLoginModal();
        this.setState({
            "email": "",
            "password": "",
            "error": false,
            "errorMessage": ""
        })
    }

    render() {
        const { ui: { loginModal, mainLoader }, handleLoginSuccess } = this.props
        return (
            <div>
                <Modal
                    isOpen={loginModal}
                    onRequestClose={this.closeModal}
                    contentLabel="Help Modal"
                    className="alertModal"
                    overlayClassName="Overlay">
                    <form>
                        <div className="login-modal">
                            <React.Fragment>                                

                                <span className="login-modal--close"  onClick={this.closeModal}></span>   

                                <h2 className="mt-30">Login to build your resume</h2>
                                <p className="error-message mt-30 text-center">
                                    {
                                        this.state.error &&
                                        <span className='pr'>{this.state.errorMessage}</span>

                                    }
                                </p>

                                <ul className="login-modal__lists">
                                    <li className="login-modal__lists--item">
                                        <input className="login-modal__input" placeholder="Email" onChange={this.handleEmailInput} autoComplete="false" />
                                    </li>
                                    <li className="login-modal__lists--item">
                                        <input className="login-modal__input" placeholder="Password" type="password" onChange={this.handlePasswordInput} autoComplete="false" />
                                    </li>
                                    {/* <li className="login-modal__lists--item text-right fs-12">
                                        <a href="#">Forget Password</a>
                                    </li> */}
                                </ul>
                                <button className="btn btn__medium btn__round btn__primary mt-20 w-100"
                                    type={'submit'} onClick={this.handleLogin}>Login
                                </button>

                                <p className="fs-12 text-center mt-30">Donâ€™t have an account yet? <a href="/register/">Register Now</a></p>
                            </React.Fragment>
                        </div>
                    </form>
                </Modal>
            </div>
        );
    }


}