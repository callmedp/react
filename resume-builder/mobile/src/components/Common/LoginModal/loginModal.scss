@import '../../../styles/variables/vars.scss';
@import '../../../styles/variables/mixins.scss';

.login-modal {
  @include square(100%);
  background-color: $white;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  padding: 20px;

  h2 {
    font-size: 2.0rem;
    color: $grey-333;
    font-weight: 700;
  }

  &__lists {
    margin-top: 30px;

    &--item {
      &:not(:last-child) {
        margin-bottom: 15px;
      }
    }
  }

  &__input {
    @include size(100%, 45px, block);
    @include border-radius(4px);
    box-shadow: 0 0 4px rgba($black, .1);
    border: 1px solid #E9E9E9;
    padding: 12px;
    font-size: 14px;
    color: $grey-666;
  }

  input::-webkit-input-placeholder {
    color: red;
  }


  &--close {
    @include size(25px, 25px, block);
    position: absolute;
    right: 0;
    top: 15px;
    transform: rotate(45deg);

    &::before,
    &::after {
      content: '';
      @include border-radius(2px);
      background-color: $grey-999;
      position: absolute;
    }

    &::before {
      @include size(20px, 2px, block);
      top: 9px;
      left: -9px;
     
    }
    &::after {
      @include size(2px, 20px, block);
    }
  }
}