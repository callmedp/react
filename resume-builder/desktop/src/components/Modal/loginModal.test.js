import { shallow } from 'enzyme';
import React from 'react';
import LoginModal from './loginModal';


describe('LoginModalTest', () => {

    it('Login Modal debug', () => {
        const  props ={
            ui: {
                loginModal: true
            }
        },
         ModalWrapperComponent = shallow(<LoginModal {...props} />);
        expect(ModalWrapperComponent).toMatchSnapshot();
    });
})

