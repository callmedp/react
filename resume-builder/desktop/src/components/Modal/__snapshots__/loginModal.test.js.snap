// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginModalTest Login Modal Test 1`] = `
<div
  className="pr"
>
  <Modal
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    className="login-modal"
    closeTimeoutMS={0}
    contentLabel="Help Modal"
    isOpen={true}
    onRequestClose={[Function]}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
  >
    <form>
      <div
        className="mt-10"
      >
        <i
          className="login-modal--close"
          onClick={[Function]}
        />
        <h2
          className="mt-30"
        >
          Login to build your resume
        </h2>
        <p
          className="login-modal--error text-center"
        />
        <ul
          className="login-modal__lists"
        >
          <li
            className="login-modal__lists--item"
          >
            <input
              className="login-modal__input"
              onChange={[Function]}
              placeholder="Email"
            />
          </li>
          <li
            className="login-modal__lists--item"
          >
            <input
              className="login-modal__input"
              onChange={[Function]}
              placeholder="Password"
              type="password"
            />
          </li>
          <li
            className="login-modal__lists--item"
          >
            <button
              className="orange-button w-100"
              onClick={[Function]}
              type="submit"
            >
              Login
            </button>
          </li>
        </ul>
        <p
          className="mt-30 mb-30 text-center"
        >
          Don’t have an account yet? 
          <a
            href="/register/"
          >
            Register Now
          </a>
        </p>
      </div>
    </form>
  </Modal>
</div>
`;

exports[`LoginModalTest Login Modal debug 1`] = `
<div
  className="pr"
>
  <Modal
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    className="login-modal"
    closeTimeoutMS={0}
    contentLabel="Help Modal"
    isOpen={true}
    onRequestClose={[Function]}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
  >
    <form>
      <div
        className="mt-10"
      >
        <i
          className="login-modal--close"
          onClick={[Function]}
        />
        <h2
          className="mt-30"
        >
          Login to build your resume
        </h2>
        <p
          className="login-modal--error text-center"
        />
        <ul
          className="login-modal__lists"
        >
          <li
            className="login-modal__lists--item"
          >
            <input
              className="login-modal__input"
              onChange={[Function]}
              placeholder="Email"
            />
          </li>
          <li
            className="login-modal__lists--item"
          >
            <input
              className="login-modal__input"
              onChange={[Function]}
              placeholder="Password"
              type="password"
            />
          </li>
          <li
            className="login-modal__lists--item"
          >
            <button
              className="orange-button w-100"
              onClick={[Function]}
              type="submit"
            >
              Login
            </button>
          </li>
        </ul>
        <p
          className="mt-30 mb-30 text-center"
        >
          Don’t have an account yet? 
          <a
            href="/register/"
          >
            Register Now
          </a>
        </p>
      </div>
    </form>
  </Modal>
</div>
`;
